docs/building-modules/keeper.md

```diff
@@ -0,0 +1,89 @@
+# Keepers
+
+## Pre-requisite Reading
+
+- [Introduction to SDK Modules](./intro.md)
+
+## Synopsis
+
+`Keeper`s refer to a Cosmos SDK abstraction whose role is to manage access to the subset of the state defined by various modules. `Keeper`s are module-specific, i.e. the subset of state defined by a module can only be accessed by a `keeper` defined in said module. If a module needs to access the subset of state defined by another module, a reference to the second module's internal `keeper` needs to be passed to the first one. 
+
+- [Motivation](#motivation)
+- [Type Definition](#type-definition)
+- [Implementing Methods](#implementing-methods)
+
+## Motivation
+
+The Cosmos SDK is a framework that makes it easy for developers to build complex decentralised applications from scratch, mainly by composing modules together. As the ecosystem of open source modules for the Cosmos SDK expands, it will become increasingly likely that some contain vulnerabilities, as a result of the negligence or malice of their developer. 
+
+The Cosmos SDK adopts a [capabilities-based approach](./ocap.md) to help developers better protect their application from unwanted inter-modules interactions, and `keeper`s are at the core of this approach. A `keeper` can be thougt quite literally as the gatekeeper of a module's store(s). Each store defined within a module comes with a `storeKey`, which grants unlimited access to it. The module's `keeper` holds this `storeKey` (which should otherwise remain unexposed), and defines [methods](#implementing-methods) for reading and writting to the store(s). 
```

Comment:
```suggestion
The Cosmos SDK adopts an [object-capabilities-based approach](./ocap.md) to help developers better protect their application from unwanted inter-module interactions, and `keeper`s are at the core of this approach. A `keeper` can be thought of quite literally as the gatekeeper of a module's store(s). Each store defined within a module comes with a `storeKey`, which grants unlimited access to it. The module's `keeper` holds this `storeKey` (which should otherwise remain unexposed), and defines [methods](#implementing-methods) for reading and writing to the store(s). 
```